build-tools/packages/version-tools/src/semver.ts

```diff
@@ -89,3 +89,52 @@ export function detectConstraintType(range: string): "minor" | "patch" {
     const maxSatisfying = semver.maxSatisfying([patch, minor], range);
     return maxSatisfying === patch ? "patch" : "minor";
 }
+
+/**
+ * Given a first and second version, returns the bump type
+ * @param v1 - The first version to compare.
+ * @param v2 - The second version to compare.
+ * @returns The bump type, or undefined if it can't be determined.
+ */
+export function detectBumpType(
+    // eslint-disable-next-line @rushstack/no-new-null
+    v1: semver.SemVer | string | null,
+    // eslint-disable-next-line @rushstack/no-new-null
+    v2: semver.SemVer | string | null,
+): VersionBumpType | undefined {
+    const v1Parsed = semver.parse(v1);
+    if (v1Parsed === null) {
+        throw new Error(`Invalid version: ${v1}`);
+    }
+
+    const v2Parsed = semver.parse(v2);
+    if (v2Parsed === null) {
+        throw new Error(`Invalid version: ${v2}`);
+    }
+
+    if (semver.compareBuild(v1Parsed, v2Parsed) >= 0) {
+        return;
```

Comment:
Is this a scenario we should guard against / report an error if we see it? This means the first version is semantically greater than the second, right? If returning undefined is the right move here, we might want to note it explicitly in the docs.

I would also recommend the following to be explicit:
```suggestion
        return undefined;
```